crates/networking/discv5/src/discovery.rs

```diff
@@ -51,65 +56,145 @@ struct QueryResult {
     find_peer_active: bool,
     pub started: bool,
+    subnets: Subnets,
 }
 
 impl Discovery {
-    pub async fn new(local_key: Keypair, config: &NetworkConfig) -> anyhow::Result<Self> {
-        let enr_local = convert_to_enr(local_key)?;
-        let enr = Enr::builder().build(&enr_local).unwrap();
+    pub async fn new(
+        local_key: libp2p::identity::Keypair,
+        config: &NetworkConfig,
+    ) -> anyhow::Result<Self> {
+        let enr_key = convert_to_enr(local_key)
+            .map_err(|e| anyhow::anyhow!("Failed to convert key: {:?}", e))?;
+        let mut enr_builder = Enr::builder();
+        enr_builder.ip(config.socket_address);
+        enr_builder.udp4(config.socket_port);
+        if let Some(attestation_bytes) = config.subnets.attestation_bytes() {
+            enr_builder.add_value::<Bytes>(ATTESTATION_BITFIELD_ENR_KEY, &attestation_bytes);
+        }
+        let enr = enr_builder
+            .build(&enr_key)
+            .map_err(|e| anyhow::anyhow!("Failed to build ENR: {:?}", e))?;
         let node_local_id = enr.node_id();
 
-        let mut discv5 = Discv5::new(enr, enr_local, config.discv5_config.clone())
-            .map_err(|err| anyhow!("Failed to create discv5: {err:?}"))?;
+        let mut discv5 = Discv5::new(enr.clone(), enr_key, config.discv5_config.clone())
+            .map_err(|e| anyhow::anyhow!("Failed to create discv5: {:?}", e))?;
 
-        // adding bootnodes to discv5
-        for enr in config.bootnodes.clone() {
-            // Skip adding ourselves to the routing table if we are a bootnode
-            if enr.node_id() == node_local_id {
+        for bootnode_enr in config.bootnodes.clone() {
+            if bootnode_enr.node_id() == node_local_id {
                 continue;
             }
-
-            if let Err(err) = discv5.add_enr(enr) {
-                error!("Failed to add bootnode to Discv5 {err:?}");
-            };
+            discv5
+                .add_enr(bootnode_enr)
+                .map_err(|e| anyhow::anyhow!("Failed to add bootnode: {:?}", e))?;
         }
 
-        // init ports
         let event_stream = if !config.disable_discovery {
             discv5
                 .start()
-                .map_err(|err| anyhow!("Failed to start discv5 {err:?}"))
-                .await?;
-            info!("Started discovery");
+                .await
+                .map_err(|e| anyhow::anyhow!("Failed to start discv5: {:?}", e))?;
+            info!("Started discovery with ENR: {:?}", discv5.local_enr());
             EventStream::Awaiting(Box::pin(discv5.event_stream()))
         } else {
             EventStream::Inactive
         };
 
         Ok(Self {
             discv5,
+            local_enr: enr,
             event_stream,
             discovery_queries: FuturesUnordered::new(),
             find_peer_active: false,
-            started: true,
+            started: !config.disable_discovery,
+            subnets: config.subnets.clone(),
```

Comment:
```suggestion
            subnets: config.subnets,
```

Reply:
Not using clone will result in compiler error here as we can't move subnets from config reference in our function.